invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/DeleteCommunityModal.js

```diff
@@ -43,8 +47,27 @@ export class DeleteCommunityModal extends Component {
 
   componentWillUnmount() {
     this.cancellableDelete && this.cancellableDelete.cancel();
+    this.cancellableMembersCountFetch && this.cancellableMembersCountFetch.cancel();
+    this.cancellableRecordsCountFetch && this.cancellableRecordsCountFetch.cancel();
   }
 
+  updateCommunityMetrics = async () => {
+    const { community } = this.props;
+    this.cancellableMembersCountFetch = withCancel(http.get(community.links.members));
+    this.cancellableRecordsCountFetch = withCancel(http.get(community.links.records));
+
+    try {
+      const membersResponse = await this.cancellableMembersCountFetch.promise;
```

Comment:
from what I understand here, you are not handling the loading state while waiting on the response, what happens between opening the modal and receiving the reponse?

Reply:
Should I set loading to `true` and not allow the user to click delete while they're being fetched?

I can also do something like 
```js
{membersCount && <Message.Content />}
```
essentially not showing the Message or Checkbox at all while the data loads.

Reply:
> Should I set loading to `true` and not allow the user to click delete while they're being fetched?

yes, you should set loading to true (check other places on how it is done)

> I can also do something like
> 
> ```js
> {membersCount && <Message.Content />}
> ```
> 
> essentially not showing the Message or Checkbox at all while the data loads.

I think it would be better to show a nice loader while the user is waiting for response (instead of not showing anything). Check other places to see how to implement the loader.

Reply:
After quite some effort to get the margins and paddings to work, I came up with this:
![image](https://user-images.githubusercontent.com/15949934/234537827-a1d1f864-8e9b-4ac2-9e67-f3ef5990a001.png)

Let me know if you think it makes sense to leave it like this. The user will rarely have to see this render, if at any time.
I'm testing this by inserting this at L61:
`await new Promise(resolve => setTimeout(resolve, 5000));`
essentially before sending the requests.

Reply:
for other reviewers: it can be tested also by setting a lower bandwidth in your browser's dev tools (f.e. to 3G)

Reply:
> After quite some effort to get the margins and paddings to work, I came up with this: ![image](https://user-images.githubusercontent.com/15949934/234537827-a1d1f864-8e9b-4ac2-9e67-f3ef5990a001.png)
> 
> Let me know if you think it makes sense to leave it like this. The user will rarely have to see this render, if at any time. I'm testing this by inserting this at ~L61~ L68: `await new Promise(resolve => setTimeout(resolve, 5000));` essentially before sending the requests.

I think it is good, as long as you set a default height of the modal to match the final rendered one, in order to prevent the UI from resizing elements once the content is loaded (it jumps a bit). It is not a blocker though, can be done in a separate PR if you prefer